File: Battleship.Server/DTO/GameDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Battleship.Server.DTO
{
    public class GameDto
    {
        [JsonProperty]
        public IList<PlayerDto> Players { get; set; }

        [JsonProperty]
        public string UrlId { get; set; }
    }
}

File: Battleship.Client/src/models/Dto/game-dto.ts
import { PlayerDto } from './player-dto'
import { JsonProperty, JsonObject } from 'json2typescript';

@JsonObject("GameDto")
export class GameDto{
    @JsonProperty('players', [PlayerDto], true)
    players: PlayerDto[] = [];

    @JsonProperty('urlId', String, true)
    token: string | undefined = undefined;
}